ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventReaderQueries.scala

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.common
+
+import anorm.{RowParser, SqlParser}
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.ledger.EventId
+import com.daml.lf.value.Value.ContractId
+import com.daml.platform.store.backend.EventStorageBackend
+import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+import com.daml.platform.store.backend.common.SimpleSqlAsVectorOf._
+import com.daml.platform.store.dao.events.Raw
+import com.daml.platform.store.dao.events.Raw.FlatEvent
+import com.daml.platform.store.interning.StringInterning
+
+import java.sql.Connection
+import scala.annotation.tailrec
+
+class EventReaderQueries(
+    queryStrategy: QueryStrategy,
+    stringInterning: StringInterning,
+) {
+  import EventStorageBackendTemplate._
```

Comment:
codestyle nit: I find the naming a little confusing, since all the rest of the backands follow a pattern except this one. Also all this here nicely fits the EventStorageBackand from the naming point of view, and from this import it is more or less natural to see all this in there.

Reply:
I was copying the naming from `TransactionPointwiseQueries` and `TransactionStreamingQueries.scala`, seems a similar use pattern.